openfasoc/generators/temp-sense-gen/tools/simulation.py

```diff
@@ -86,6 +86,10 @@ def generate_runs(
                 error_data = calculate_sim_error(sim_output_lines=sim_output_file.readlines())
                 all_result_file.write("\n".join(error_data))
 
+        # dump final simulation state to log file called sim_state_file
+        with open(spiceDir + "/sim_state_file.txt", 'w') as sim_state_file:
```

Comment:
It would be better to write this file in the run directory instead since there could be multiple runs and each would have a different output.

Reply:
This file is only supposed to write out the completed, final and ongoing sims. It doesn't really have anything to do with the intermediate runs

Reply:
Sorry, I worded that incorrectly. What I meant by a "run" was one particular configuration and all of its simulations (eg: 4 inverters, 6 headers and their 7 simulation outputs). The run directory is `simulations/run/prePEX4inv6head`). It would be better to store this file inside this directory rather than the `spiceDir`. Alternatively, you could name the file `sim_state_prePEX_4inv_6head` so that multiple results could be stored.

Reply:
Ill look into that. Thanks Harsh!

Reply:
So I looked through this and it seems like after `check_search_done()` is called, the `ninv` and `nhead` variables are just integers rather than lists, so the actual "lists" of header and inverter configs being passed is just one set (of optimum configs)
this must also be why only one directory is created inside the `run` folder